bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/java/org/eclipse/smarthome/automation/module/core/internal/TriggeredRunningStateTest.java

```diff
@@ -0,0 +1,281 @@
+        Assert.assertNotNull(myLampItemAction2);
+        waitForAssert(() -> {
+            Assert.assertEquals(OnOffType.ON, myLampItemAction2.getState());
+        });
+
+    }
+
+    @Test
+    public void testTriggeredState() throws ItemNotFoundException, InterruptedException {
+
+        final RuleRegistry ruleRegistry = getService(RuleRegistry.class);
+        Rule rule = prepareRule();
+        ruleRegistry.add(rule);
+        ruleRegistry.setEnabled(rule.getUID(), true);
+
+        waitForAssert(() -> {
+            Assert.assertEquals(RuleStatus.IDLE, ruleRegistry.getStatusInfo(rule.getUID()).getStatus());
+        });
+
+        // prepare condition item state - the condition must be unsatisfied
+        final ItemRegistry itemRegistry = getService(ItemRegistry.class);
+        SwitchItem myLampItem = (SwitchItem) itemRegistry.getItem("myLampItemCondition");
+        myLampItem.send(OnOffType.OFF);
+
+        Assert.assertNotNull(myLampItem);
+        waitForAssert(() -> {
+            Assert.assertEquals(OnOffType.OFF, myLampItem.getState());
+        });
+
+        // prepare action item state
+        final SwitchItem myLampItemAction = (SwitchItem) itemRegistry.getItem("myLampItemAction");
+        myLampItemAction.send(OnOffType.OFF);
+
+        Assert.assertNotNull(myLampItemAction);
+        waitForAssert(() -> {
+            Assert.assertEquals(OnOffType.OFF, myLampItemAction.getState());
+        });
+
+        // Test rule
+        SwitchItem myMotionItem = (SwitchItem) itemRegistry.getItem("myMotionItemTrigger");
+
+        myMotionItem.send(OnOffType.ON);
+
+        boolean ruleStatusInfoEventFound = false;
+        boolean triggeredEventFound = false;
+        String ruleStatusTopic = "smarthome/rules/{ruleID}/state".replace("{ruleID}", rule.getUID());
+        long wait = TimeUnit.MILLISECONDS.toNanos(DFL_TIMEOUT);
+        final long nanoEnd = System.nanoTime() + wait;
+        do {
+            final Event event = events.poll(wait, TimeUnit.NANOSECONDS);
+
+            Assert.assertNotNull(event);
+            wait = nanoEnd - System.nanoTime();
+
+            if (ruleStatusTopic.equals(event.getTopic())) {
+                if (!(event instanceof RuleStatusInfoEvent)) {
+                    continue;
+                }
+
+                final RuleStatusInfoEvent ruleStateEvent = (RuleStatusInfoEvent) event;
+                switch (ruleStateEvent.getStatusInfo().getStatus()) {
+                    case TRIGGERED:
+                        triggeredEventFound = true;
+                        break;
+                    case IDLE:
+                        ruleStatusInfoEventFound = triggeredEventFound;
+                        break;
+                }
+            }
+        } while (!ruleStatusInfoEventFound && wait > 0);
+        Assert.assertTrue(ruleStatusInfoEventFound);
+        Assert.assertNotNull(myLampItemAction);
+        Assert.assertEquals(OnOffType.OFF, myLampItemAction.getState());
```

Comment:
Actually no, because the condition is not satisfied (myLampCondition is OFF) and the rule is not executed.  Therefor the action myLampItemAction is not executed and the state is not changed.